command/s3_put_test.go

```diff
@@ -302,6 +304,55 @@ func TestExpandS3PutParams(t *testing.T) {
 	})
 }
 
+func TestSignedUrlVisibility(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	for _, vis := range []string{"signed", "private"} {
+		var err error
+		s := s3put{
+			AwsKey:        "key",
+			AwsSecret:     "secret",
+			Bucket:        "bucket",
+			BuildVariants: []string{},
+			ContentType:   "content-type",
+			Permissions:   s3.BucketCannedACLPublicRead,
+			RemoteFile:    "remote",
+			Visibility:    vis,
+		}
+
+		comm := client.NewMock("http://localhost.com")
+		conf := &model.TaskConfig{
+			Expansions:   &util.Expansions{},
+			Task:         &task.Task{Id: "mock_id", Secret: "mock_secret"},
+			Project:      &model.Project{},
+			BuildVariant: &model.BuildVariant{},
+		}
+		logger, err := comm.GetLoggerProducer(ctx, client.TaskData{ID: conf.Task.Id, Secret: conf.Task.Secret}, nil)
+		require.NoError(t, err)
+
+		localFiles := []string{"file1", "file2"}
+		remoteFile := "remote file"
+
+		require.NoError(t, s.attachFiles(ctx, comm, logger, localFiles, remoteFile))
+
+		attachedFiles := comm.AttachedFiles
+		assert := assert.New(t)
```

Comment:
the evergreen codebase is kinda dodgy on this account, but it's more clear if you pass t to each assert rather than using assert.New()